jets/e/secp.c

```diff
@@ -0,0 +1,344 @@
+/* j/5/secp.c
+**
+*/
+#include "all.h"
+#include "../include/secp256k1.h"
+#include "../include/secp256k1_recovery.h"
+
+// end-to-end test:
+//   =hash 123
+//   =priv 99
+//   =nonce (make-k:secp256k1:secp:crypto `@uvI`hash priv)
+//   =sig (ecdsa-raw-sign:secp256k1:secp:crypto `@uvI`hash priv)
+//   =reco (ecdsa-raw-recover:secp256k1:secp:crypto `@uvI`hash v:sig  r:sig s:sig)
+
+
+
+
+//------------------------------------------------------------
+//----- util funcs
+//------------------------------------------------------------
+
+#if 0
+static void pretty_print_hex(char * headline,
+                             unsigned char * ptr,
+                             unsigned int num_bytes)
+{
+  fprintf(stderr, "\r%s:\n\r  ", headline);
+  for (int ii=0 ; ii < num_bytes; ii++) {
+    fprintf(stderr, "%02x", ptr[ii]);
+    if (0 == (ii + 1) % 16){
+          fprintf(stderr, "\n\r  ");
+    } else {
+          fprintf(stderr, ", ");
+    }
+  }
+  fprintf(stderr, "\n\r");
+
+}
+#endif
+
+// no guarantees if 'in' and 'out' overlap / are the same
+static void byte_reverse(c3_y *in,
+                         c3_y *out,
+                         size_t n_bytes)
+{
+  size_t ii;
+  for (ii = 0; ii < n_bytes; ii++){
+    out[n_bytes - 1 - ii] = in[ii];
+  }
+
+  return;
+}
+
+// identical to u3r_bytes, but reverses bytes in place.
+// could be cleaner if we modified u3r_bytes(), but not gonna do that
```

Comment:
I think it would be fine to copy the `u3r_bytes()` implementation, and then make the relevant changes so as to be big-endian. If you do, that should probably go in `retrieve.c`. This looks fine here, my only nitpick is that the name could be confusing (as it implies that the function *is* in `retrieve.c`).

Reply:
Comment added to explain why it exists.